jms/src/test/java/docs/javadsl/JmsConnectorsTest.java

```diff
@@ -868,6 +869,73 @@ public void passThroughEmptyMessageEnvelopes() throws Exception {
         });
   }
 
+  @Test
+  public void requestReplyWithTempQueues() throws Exception {
+    withServer(
+        server -> {
+          ConnectionFactory connectionFactory = server.createConnectionFactory();
+
+          Connection connection = connectionFactory.createConnection();
+          connection.start();
+          Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+          TemporaryQueue tempQueue = session.createTemporaryQueue();
+          Destination tempQueueDest =
+              akka.stream.alpakka.jms.Destination.createDestination(tempQueue);
+          String message = "ThisIsATest";
+          Function<String, String> reverse = (in) -> new StringBuilder(in).reverse().toString();
+          String correlationId = UUID.randomUUID().toString();
+
+          Sink<JmsTextMessage, CompletionStage<Done>> toRespondSink =
+              JmsProducer.sink(
+                  JmsProducerSettings.create(system, connectionFactory).withQueue("test"));
+
+          CompletionStage<Done> toRespondStreamCompletion =
+              Source.single(
+                      JmsTextMessage.create(message)
+                          .withHeader(JmsCorrelationId.create(correlationId))
+                          .withHeader(new JmsReplyTo(tempQueueDest)))
+                  .runWith(toRespondSink, materializer);
+
+          // #request-reply
+          JmsConsumerControl respondStreamControl =
+              JmsConsumer.create(
+                      JmsConsumerSettings.create(system, connectionFactory).withQueue("test"))
+                  .map(JmsMessage::create)
+                  .collectType(JmsTextMessage.class)
+                  .map(
+                      textMessage -> {
+                        JmsTextMessage m = JmsTextMessage.create(reverse.apply(textMessage.body()));
+                        for (JmsHeader h : JavaConverters.asJavaCollection(textMessage.headers()))
```

Comment:
Oh, I wasn't aware that there is no Java API to access the message headers, properties and destination. We should add that in another change (see https://github.com/akka/alpakka/issues/1800).